src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc

```diff
@@ -0,0 +1,526 @@
+  }
+
+  ~SetAudioStreamingIndicatorRequestTest() {
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
+  }
+
+  MessageSharedPtr CreateFullParamsSO() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    (*msg)[am::strings::msg_params] = msg_params;
+    return msg;
+  }
+
+  void SetUp() OVERRIDE {
+    mock_app_ = CreateMockApp();
+    ON_CALL(app_mngr_, application(kConnectionKey))
+        .WillByDefault(Return(mock_app_));
+    ON_CALL(*mock_app_, app_id()).WillByDefault(Return(kConnectionKey));
+    ON_CALL(app_mngr_, hmi_interfaces())
+        .WillByDefault(ReturnRef(hmi_interfaces_));
+  }
+
+  void TearDown() OVERRIDE {
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
+  }
+
+  void DefineInterfaceAvailable(
+      const am::HmiInterfaces::InterfaceID interface) {
+    ON_CALL(hmi_interfaces_, GetInterfaceState(interface))
+        .WillByDefault(Return(am::HmiInterfaces::STATE_AVAILABLE));
+  }
+
+  void DefineInterfaceNotAvailable(
+      const am::HmiInterfaces::InterfaceID interface) {
+    ON_CALL(hmi_interfaces_, GetInterfaceState(interface))
+        .WillByDefault(Return(am::HmiInterfaces::STATE_NOT_AVAILABLE));
+  }
+
+  void DefineHMILevelUIAvailable() {
+    ON_CALL(*mock_app_, hmi_level())
+        .WillByDefault(Return(mobile_apis::HMILevel::HMI_FULL));
+    ON_CALL(hmi_interfaces_, GetInterfaceFromFunction(_))
+        .WillByDefault(Return(am::HmiInterfaces::HMI_INTERFACE_UI));
+    DefineInterfaceAvailable(am::HmiInterfaces::HMI_INTERFACE_UI);
+  }
+
+  void VerifyCommandResults(MessageSharedPtr msg,
+                            const mobile_apis::Result::eType mobile_code,
+                            const bool is_success) {
+    EXPECT_EQ((*msg)[am::strings::msg_params][am::strings::success].asBool(),
+              is_success);
+    EXPECT_EQ((*msg)[am::strings::msg_params][am::strings::result_code].asInt(),
+              mobile_code);
+  }
+
+  void AppSetup(const bool is_navi,
+                const bool is_voice_communication,
+                const bool is_media) {
+    ON_CALL(*mock_app_, is_navi()).WillByDefault(Return(is_navi));
+    ON_CALL(*mock_app_, is_media_application()).WillByDefault(Return(is_media));
+    ON_CALL(*mock_app_, is_voice_communication_supported())
+        .WillByDefault(Return(is_voice_communication));
+  }
+
+  void MediaAppSetup() {
+    AppSetup(kIsNotNavi, kIsNotVoiceCommunication, kIsMedia);
+  }
+
+  void VoiceCommunicationAppSetup() {
+    AppSetup(kIsNotNavi, kIsVoiceCommunication, kIsNotMedia);
+  }
+
+  void NaviAppSetup() {
+    AppSetup(kIsNavi, kIsNotVoiceCommunication, kIsNotMedia);
+  }
+
+  void IncorrectAppSetup() {
+    AppSetup(kIsNotNavi, kIsNotVoiceCommunication, kIsNotMedia);
+  }
+
+  void TestCommandRunMethod(
```

Comment:
@tkireva, `TestCommandRunMethod()` and `TestCommandOnEventCall()` kind of functions already included to the `ComandRequestTest` class. Please use them instead(`CatchMobileCommandResult()` and `CatchHMICommandResult()`).